build.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/sh
+
+set -eu
+
+ARCH="${ARCH:-armhf}"
+CUR_DIR="$(pwd)"
+BUILD_DIR="${CUR_DIR}/.build_${ARCH}"
+DEBIAN_VERSION="${DEBIAN_VERSION:-jessie}"
+ROOTFS_ARCHIVE="rootfs.tar.xz"
+ROOTFS_DIR="${BUILD_DIR}/rootfs"
+QEMU_STATIC_BIN="$(command -v qemu-arm-static)"
+
+
+cleanup()
+{
+    local mount_points
+    mount_points="$(mount | grep "${ROOTFS_DIR}")"
+
+    if [ "${mount_points}" = "" ]; then
+        if [ -d "${BUILD_DIR}" ]; then
+            rm -rf "${BUILD_DIR}"
```

Comment:
your core step of the cleanup step is deeply nested and hidden, personally i prefer to avoid nesting by fail often and fail early.